redis.conf

```diff
@@ -1490,16 +1490,21 @@ lua-time-limit 5000
 #
 # In order to make Redis Cluster working in such environments, a static
 # configuration where each node knows its public address is needed. The
-# following two options are used for this scope, and are:
+# following four options are used for this scope, and are:
 #
 # * cluster-announce-ip
 # * cluster-announce-port
+# * cluster-announce-tls-port
 # * cluster-announce-bus-port
 #
-# Each instructs the node about its address, client port, and cluster message
-# bus port. The information is then published in the header of the bus packets
-# so that other nodes will be able to correctly map the address of the node
-# publishing the information.
+# Each instructs the node about its address, client ports (for connections
+# without and with TLS) and cluster message bus port. The information is then
+# published in the header of the bus packets so that other nodes will be able to
+# correctly map the address of the node publishing the information.
+#
+# If cluster-tls is set to yes and cluster-announce-tls-port is omitted or set
+# to zero, then cluster-announce-port refers to the TLS port. Note also that
+# cluster-announce-tls-port has no effect if cluster-tls is set to no.
```

Comment:
This is according to @oranagra's idea. (I didn't mention "historical reasons"). What do you think?

Reply:
@zuiderkwast
So basically the cluster bus message should always specify `port` and an optional `secondary-port`, and we should always include both (if configured). Does that make sense?

Reply:
@yossigo Yes, in the cluster bus, this is the backward compatible way.

I haven't added TLS as a secondary port in a non-TLS cluster (non-TLS cluster bus). Are you OK with postponing that as future work? (My estimate is that it's easy to implement, but some more work is needed in the test framework.)

Reply:
I don't think we really ever need to support the TLS as a secondary port. I know you asked for it, but this seems like a dubious production configuration to have.

Reply:
@madolson Why do you think so? The industry is moving towards TLS everywhere but the real security value is not always great, and in some cases the cost can be high.

Consider this example:
* A bunch of RaspberryPI physically interconnected through a small network switch
* Serving different types of clients through a wireless network (some IoT deployment)

TLS on the cluster bus may have high impact, and low security value.
TLS for clients has high value, but some may not support it and require plain fallback.

I think we sometimes forget how versatile Redis deployments are.

Reply:
... or a cluster deployed on a single machine, or even in a single container, just to make use of multiple CPUs.

Reply:
Sorry, there was a lot of context to catch up and I misunderstood what the suggestion was. I think having TLS for outbound ports but running the clusterbus using TCP makes plenty of sense.